src/Security/Authentication/Certificate/src/CertificateForwarderMiddleware.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication.Certificate
+{
+
+    // TODO: move to header forwarder
+    public class CertificateForwarderMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly CertificateForwarderOptions _options;
+        private readonly ILogger _logger;
+
+        public CertificateForwarderMiddleware(
```

Comment:
This should not be a new middleware, it should go into the existing FowardedHeaders middleware as another optional header.

Reply:
You mean the HeaderPropagation middleware?  That isn't in the shared framework currently when I tried to reference it from security it said nope...

Reply:
No, this one:
https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

Reply:
Alright as we discussed, moved the CertForwarderMiddleware stuff in to HttpOverrides.

What do you guys think about getting rid of the AddCertificateForwader method on configure services and switching to just having that only as part of UseCertificateForwarder()?  Do we really need to allow them to configure the options in DI?  Can still take the Lambda, but then we wouldn't need two methods.

Would just be a single call in Configure
```
app.UseCertificateForwarder(options => options.CertificateHeader = headerName);
```

Reply:
We've avoided adding any new middleware extensions that passes options to UseXx. It's a lot cleaner to keep all of the config in ConfigureServices. AddCertificateForwader is primarily for discoverability.

Reply:
Yeah I only asked because the other two had UseXx that took options :)